trollmoves/filescleaner.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025 Pytroll Developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utility functions for cleaning files and directories."""
+
+
+import datetime as dt
+import logging
+import os
+from glob import glob
+
+from posttroll.message import Message
+
+LOGGER = logging.getLogger(__name__)
+
+
+class FilesCleaner():
+    """Files cleaner class to accomodate cleaning of files acording to configured rules."""
+
+    def __init__(self, publisher, section, info, dry_run=True):
+        """Initialize the class."""
+        self.pub = publisher
+        self.section = section
+        self.info = info
+        self.dry_run = dry_run
+        self.stat_time_method = self.info.get('st_time', 'st_ctime')
```

Comment:
Should the setting name in `self.info` match the variable name? Might be more clear for the user. Later on changing the name would be a breaking change.

Reply:
Yes, good point. I think `stat_time_method` is better than `st_time`. So will change!